src/passes/inlineAssemblyTransformer.ts

```diff
@@ -0,0 +1,124 @@
+import { ASTMapper } from '../ast/mapper';
+import { AST } from '../ast/ast';
+import {
+  createExpressionStatement,
+  createIdentifier,
+  createNumberLiteral,
+  createTuple,
+} from '../utils/nodeTemplates';
+
+import {
+  InlineAssembly,
+  YulNode,
+  Identifier,
+  ASTNode,
+  BinaryOperation,
+  Assignment,
+  Block,
+  Expression,
+  ExpressionStatement,
+  VariableDeclaration,
+  UncheckedBlock,
+  Literal,
+  Statement,
+} from 'solc-typed-ast';
+import assert from 'assert';
+import { WillNotSupportError } from '../export';
+
+class YulTransformer {
+  vars: Map<string, VariableDeclaration>;
+  assemblyRoot: InlineAssembly;
+  ast: AST;
+
+  constructor(ast: AST, assemblyRoot: InlineAssembly) {
+    this.ast = ast;
+    this.assemblyRoot = assemblyRoot;
+    const blockRefIds = assemblyRoot.externalReferences.map((ref) => ref.declaration);
+    const blockRefs = [...assemblyRoot.context!.map].filter(([id, _]) =>
+      blockRefIds.includes(id),
+    ) as [number, VariableDeclaration][];
+    this.vars = new Map(blockRefs.map(([id, ref]) => [ref.name, ref]));
+  }
+
+  run(node: YulNode): Statement {
+    const methodName = `create${node.nodeType}`;
+    if (!(methodName in this))
+      throw new WillNotSupportError(`${node.nodeType} is not supported`, this.assemblyRoot, false);
+    const method = this[methodName as keyof YulTransformer] as (node: YulNode) => ASTNode;
+    return method.bind(this)(node);
+  }
+
+  createYulLiteral(node: YulNode): Literal {
+    return createNumberLiteral(node.value, this.ast);
+  }
+
+  createYulIdentifier(node: YulNode): Identifier {
+    return createIdentifier(this.vars.get(node.name)!, this.ast);
+  }
+
+  createYulFunctionCall(node: YulNode): BinaryOperation {
+    const binaryOps: { [name: string]: string } = { add: '+', mul: '*', div: '/', sub: '-' };
+    if (Object.keys(binaryOps).includes(node.functionName.name)) {
```

Comment:
its enough to check if the result of `binaryOps[node.functionName.name]` is not undefined